admin/adminChatAnalyser.php

```diff
@@ -0,0 +1,201 @@
+<?php
+
+$countryID1 = $countryID2 = array();
+$gameID=$userID=0;
+$reason = 'Press was checked for gameID: ';
+
+if (!defined('IN_CODE'))
+{
+	chdir ('..');
+	require_once('header.php');
+}
+
+if (!$User->type['Moderator'])
+	die ('Only admins or mods can run this script');
+
+$asTxt = (isset($_REQUEST['TXT'])?true:false);
+$asCsv = (isset($_REQUEST['CSV'])?true:false);
+
+if ( isset($_REQUEST['gameID']) )
+	$gameID=(int)$_REQUEST['gameID'];
+	
+if ( isset($_REQUEST['userID']) && $gameID == 0 )
+	$userID=(int)$_REQUEST['userID'];
+	
+if ( isset($_REQUEST['countryID1']) )
+	foreach ($_REQUEST['countryID1'] as $user)
+		$countryID1[]=(int)$user;
+
+if ( isset($_REQUEST['countryID2']) )
+	foreach ($_REQUEST['countryID2'] as $user)
+		$countryID2[]=(int)$user;
+
+if ( isset($_REQUEST['reason']) && $gameID == 0 )
+	$reason = htmlentities( $_REQUEST['reason'], ENT_NOQUOTES, 'UTF-8');
+
+// Output as Txt or Csv:
+if ((($gameID != 0) && (count($countryID1) > 0) && (count($countryID2) > 0)) && ($asTxt OR $asCsv))
+{
+	$Variant=libVariant::loadFromGameID($gameID);
+	
+	$sql = 'SELECT message, toCountryID, fromCountryID, turn, timeSent
+				FROM wD_GameMessages WHERE
+					gameID = '.$gameID.' AND
+					(
+						( toCountryID IN ('.implode(', ',$countryID1).') AND fromCountryID IN ('.implode(', ',$countryID2).') )
+							OR
+						( toCountryID IN ('.implode(', ',$countryID2).') AND fromCountryID IN ('.implode(', ',$countryID1).') )
+					)
+				ORDER BY id ASC';
+	
+	$tabl = $DB->sql_tabl($sql);
+	
+	
+	if ($asCsv)
+	{
+		header("Content-type: csv/plain; charset=utf-8");
+		header("Content-disposition: inline; filename=Chatlog.csv");
+		print '"timeSent";"gameDate";"To";"From";"Message"'."\n";
+	}
+	else
+	{
+		header("Content-type: txt/plain; charset=utf-8");
+		header("Content-disposition: inline; filename=Chatlog.txt");
+	}
+		
+	while ( list($message, $to, $from, $turn, $timeSent) = $DB->tabl_row($tabl) )
+	{
+		if ($asCsv)
+			print '"'.gmstrftime("%d %b %y %I:%M %p", $timeSent).'";'.
+					'"'.$Variant->turnAsDate($turn).'";'.
+					'"'.($to==0?'Global':$Variant->countries[$to-1]).'";'.
+					'"'.($from==0?'Global':$Variant->countries[$from-1]).'";'.
+					'"'.addslashes($message).'"'."\n";		
+		else
+			print gmstrftime("%d %b %y %I:%M %p", $timeSent).
+					' ('.$Variant->turnAsDate($turn).')'.
+					' / To '.($to==0?'Global':$Variant->countries[$to-1]).
+					' from '.($from==0?'Global':$Variant->countries[$from-1]).': '.
+					$message."\r\n";		
+	}
+	exit(0);
```

Comment:
I think this whole function could be pulled out to a separate file, and called in an If. The exit call would then not be necessary.